app/models/pusher.rb

```diff
@@ -3,22 +3,29 @@
 class Pusher
   attr_reader :user, :spec, :message, :code, :rubygem, :body, :version, :version_id, :size
 
-  def initialize(user, body, remote_ip = "")
+  def initialize(user, body, remote_ip = "", scoped_rubygem = nil)
     @user = user
     @body = StringIO.new(body.read)
     @size = @body.size
     @indexer = Indexer.new
     @remote_ip = remote_ip
+    @scoped_rubygem = scoped_rubygem
   end
 
   def process
-    pull_spec && find && authorize && verify_mfa_requirement && validate && save
+    pull_spec && find && authorize && verify_gem_scope && verify_mfa_requirement && validate && save
   end
 
   def authorize
     rubygem.pushable? || rubygem.owned_by?(user) || notify_unauthorized
   end
 
+  def verify_gem_scope
+    return true unless @scoped_rubygem && rubygem != @scoped_rubygem
+
+    notify("Rubygem cannot be scoped to this API key", 403)
```

Comment:
I think this should say `Gem` or `RubyGem`.

Reply:
Yeah the wording on this message could be improved. How does "This API key cannot perform this action on this gem." sound?